app/controllers/api/v1/packages_controller.rb

```diff
@@ -171,6 +170,16 @@ def designate_partial_item
         end
       end
 
+      def split_package
+        qty_to_split = package_params[:quantity].to_i
+        if not_splittable?(qty_to_split)
+          render json: { errors: I18n.t('package.split_qty_error', qty: @package.quantity) }, status: 422
```

Comment:
This can be done at model level. instead of adding error from here.

Reply:
@namrataukirde , That's how it is done for `custom error messages` on other methods so I just followed the convention.

Reply:
Okay, May be we can do this later when we will move code to service class. :)